components/edit_post_modal/edit_post_modal.tsx

```diff
@@ -249,60 +311,83 @@ class EditPostModal extends React.PureComponent {
-        });
-    }
+        this.setState(
+            {
+                editText: message,
+                caretPosition: newCaretPosition,
+            },
+            () => {
+                if (this.editbox) {
+                    Utils.setCaretPosition(this.editbox.getInputBox(), newCaretPosition);
+                }
+            },
+        );
+    };
 
-    handleEditKeyPress = (e) => {
+    handleEditKeyPress = (e: React.KeyboardEvent<Element>) => {
         const {ctrlSend, codeBlockOnCtrlEnter} = this.props;
 
-        const {allowSending, ignoreKeyPress} = postMessageOnKeyPress(e, this.state.editText, ctrlSend, codeBlockOnCtrlEnter, Date.now(), this.lastChannelSwitchAt, this.state.caretPosition);
+        const {allowSending, ignoreKeyPress} = postMessageOnKeyPress(e, this.state.editText, ctrlSend, codeBlockOnCtrlEnter, Date.now(), this.lastChannelSwitchAt, this.state.caretPosition) as {allowSending: boolean; ignoreKeyPress?: boolean};
 
         if (ignoreKeyPress) {
             e.preventDefault();
             e.stopPropagation();
             return;
         }
 
-        if (allowSending) {
+        if (allowSending && this.editbox) {
             e.preventDefault();
             this.editbox.blur();
             this.handleEdit();
         }
-    }
+    };
 
-    handleMouseUpKeyUp = (e) => {
+    handleMouseUpKeyUp = (e: React.MouseEvent<Element, MouseEvent> | React.KeyboardEvent<Element>) => {
         const caretPosition = Utils.getCaretPosition(e.target);
         this.setState({
             caretPosition,
         });
-    }
+    };
 
-    handleSelect = (e) => {
-        Utils.adjustSelection(this.editbox.getInputBox(), e);
-    }
+    handleSelect = (
+        e: React.SyntheticEvent<Element, Event> | React.SyntheticEvent<Modal, Event>,
+    ) => {
+        if (this.editbox) {
+            Utils.adjustSelection(this.editbox.getInputBox(), e);
+        }
+    };
 
-    handleKeyDown = (e) => {
+    handleKeyDown = (
+        e: React.KeyboardEvent<Element> | React.KeyboardEvent<Modal>,
+    ) => {
         const {ctrlSend, codeBlockOnCtrlEnter} = this.props;
 
         const ctrlOrMetaKeyPressed = e.ctrlKey || e.metaKey;
         const ctrlKeyCombo = Utils.cmdOrCtrlPressed(e) && !e.altKey && !e.shiftKey;
         const ctrlAltCombo = Utils.cmdOrCtrlPressed(e, true) && e.altKey;
-        const ctrlEnterKeyCombo = (ctrlSend || codeBlockOnCtrlEnter) && Utils.isKeyPressed(e, KeyCodes.ENTER) && ctrlOrMetaKeyPressed;
-        const markdownHotkey = Utils.isKeyPressed(e, KeyCodes.B) || Utils.isKeyPressed(e, KeyCodes.I);
+        const ctrlEnterKeyCombo =
+      (ctrlSend || codeBlockOnCtrlEnter) &&
+      Utils.isKeyPressed(e, KeyCodes.ENTER) &&
+      ctrlOrMetaKeyPressed;
+        const markdownHotkey =
+      Utils.isKeyPressed(e, KeyCodes.B) || Utils.isKeyPressed(e, KeyCodes.I);
         const markdownLinkKey = Utils.isKeyPressed(e, KeyCodes.K);
```

Comment:
```suggestion
        const ctrlEnterKeyCombo =
            (ctrlSend || codeBlockOnCtrlEnter) &&
            Utils.isKeyPressed(e, KeyCodes.ENTER) &&
            ctrlOrMetaKeyPressed;
        const markdownHotkey =
            Utils.isKeyPressed(e, KeyCodes.B) ||
            Utils.isKeyPressed(e, KeyCodes.I);
        const markdownLinkKey =
            Utils.isKeyPressed(e, KeyCodes.K);
```
nit: formatting/consistency